packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/appmesh.ts

```diff
@@ -259,17 +259,33 @@ export class AppMeshExtension extends ServiceExtension {
       throw new Error('You must add a CloudMap namespace to the ECS cluster in order to use the AppMesh extension');
     }
 
+    function addListener(protocol: appmesh.Protocol, port: number) {
+      if (protocol === appmesh.Protocol.HTTP2) {
```

Comment:
Can you use a `switch` instead?

```ts
switch (protocol) {
  case appmesh.Protocol.HTTP2: return appmesh.VirtualNodeListener.http2NodeListener({ port });
  // ...
}
```

This way, the compiler will guarantee that all enum cases are covered.